synapse/lib/cli.py

```diff
@@ -222,26 +206,30 @@ def printf(self, mesg, addnl=True):
                                name='runCmdOpts')
 
 
-class Cli(s_eventbus.EventBus):
+class Cli(s_base.Base):
     '''
     A modular / event-driven CLI base object.
     '''
-    def __init__(self, item, outp=None, **locs):
-        s_eventbus.EventBus.__init__(self)
+    async def __anit__(self, item, outp=None, **locs):
+
+        await s_base.Base.__anit__(self)
 
         if outp is None:
             outp = s_output.OutPut()
 
         self.outp = outp
         self.locs = locs
+
+        path = s_common.getSynPath('cmdr_history')
+
+        hist = FileHistory(path)
```

Comment:
this is not okay since a unsafe test use of this object could muck up a real users cmdr_history

Reply:
Good catch. Maybe we could/would/should make the test setup override something to redirect getSynPath() rather than having this problem everywhere?

Reply:
previously we were lazily-loading the history when we were entering into the cmd loop; and only dealing with the actual file history *if* a local variable was set (which was set in the cmdr main()).  We could adopt that strategy again; but i don't know enough about prompt-toolkit ot know if we can delay the self.session initialization or not.

Reply:
Moved session construction into the prompt() method.